File: src/minecraft/net/machinemuse/api/electricity/ElectricConversions.scala
package net.machinemuse.api.electricity

import net.minecraft.item.ItemStack
import net.machinemuse.powersuits.common.ModCompatability
import universalelectricity.core.electricity.ElectricityPack
import net.machinemuse.api.ModuleManager

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 4:51 AM, 4/28/13
 */
object ElectricConversions {
  // IC2
  final val IC2_TIER: String = "IC2 Tier"

  def getTier(stack: ItemStack): Int = ModuleManager.computeModularProperty(stack, IC2_TIER).toInt

  def museEnergyToEU(museEnergy: Double): Double = museEnergy / ModCompatability.getIC2Ratio

  def museEnergyFromEU(eu: Double): Double = eu * ModCompatability.getIC2Ratio

  // TE
  def museEnergyToRF(museEnergy: Double): Int = Math.ceil(museEnergy / ModCompatability.getRFRatio).toInt

  def museEnergyFromRF(rf: Int): Double = rf * ModCompatability.getRFRatio

  // BC
  def museEnergyToMJ(museEnergy: Double): Double = museEnergy / ModCompatability.getBCRatio

  def museEnergyFromMJ(mj: Double): Double = mj * ModCompatability.getBCRatio

  // UE
  def museEnergyToElectricityPack(museEnergy: Double, voltage: Float): ElectricityPack = new ElectricityPack(museEnergyToJoules(museEnergy) * 20.0F / voltage, voltage)

  def museEnergyFromElectricityPack(pack: ElectricityPack): Double = museEnergyFromJoules(pack.amperes / 20.0F * pack.voltage)

  def museEnergyToJoules(energy: Double): Float = (energy / ModCompatability.getUERatio).toFloat

  def museEnergyFromJoules(energy: Float): Double = energy * ModCompatability.getUERatio
}

File: src/main/java/net/machinemuse/numina/api/energy/ElectricConversions.java
package net.machinemuse.numina.api.energy;

import net.machinemuse.numina.api.module.ModuleManager;
import net.machinemuse.numina.common.config.NuminaConfig;
import net.machinemuse.powersuits.common.ModCompatibility;
import net.machinemuse.utils.ElectricItemUtils;
import net.minecraft.item.ItemStack;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 4:51 AM, 4/28/13
 *
 * Ported to Java by lehjr on 11/4/16.
 */
public final class ElectricConversions {



    /* Thermal Expansion -------------------------------------------------------------------------- */
    public static int museEnergyToRF(final double museEnergy) {
        return (int)Math.ceil(museEnergy / ModCompatibility.getRFRatio());
    }

    public static double museEnergyFromRF(final int rf) {
        return rf * ModCompatibility.getRFRatio();
    }









    /** Industrialcraft 2 -------------------------------------------------------------------------- */
    public static final String IC2_TIER = "IC2 Tier";

    public static int getTier(final ItemStack stack) {
        return ElectricItemUtils.getTierForItem(stack);
    }

    public static double museEnergyToEU(final double museEnergy) {
        return museEnergy / NuminaConfig.getIC2Ratio();
    }

    public static int museEnergyFromEU(final double eu) {
        return (int) Math.round(eu * NuminaConfig.getIC2Ratio());
    }

    /** Mekanism ------------------------------------------------------------------------------------ */
    public static double museEnergyToMek(final double museEnergy) {
        return Math.ceil(museEnergy / NuminaConfig.getMekRatio());
    }

    public static int museEnergyFromMek(final double mj) {
        return (int) Math.round(mj * NuminaConfig.getMekRatio());
    }

    /** Applied Energistics 2 --------------------------------------------------------------------- */
    public static double museEnergyToAE(final double museEnergy) {
        return museEnergy / NuminaConfig.getAE2Ratio();
    }

    public static int museEnergyFromAE(final double ae) {
        return (int) Math.round(ae * NuminaConfig.getAE2Ratio());
    }
}